docs/releasing.md

```diff
@@ -1,116 +1,141 @@
 
-   ```bash
-   git tag -a -s v0.x.y -m v0.x.y
-   git tag -a -s test/v0.x.y -m test/v0.x.y
-   ```
+This triggers two things:
 
-1. Push the tags to the GitHub repository:
+- Release workflow on Github side that creates a draft release with correct
+  content, comparing the tag to previous tag
+- Quay starts building release image with the release tag
 
-   ```bash
-   git push origin v0.x.y
-   git push origin test/v0.x.y
-   ```
+We also need to do one or more tags for Go ecosystem:
 
-   NB: `origin` should be the name of the remote pointing to
-   `github.com/metal3-io/cluster-api-provider-metal3`.
-1. Check the tags are existent in upstream repository. If they are,
-   GitHub workflow should have automatically created draft releases
-   for you. Please make sure the release notes are checked
-   thoroughly before they are published.
+- For any directory with `go.mod` in it (excluding `hack/tools`), create another
+  Git tag with directory prefix, ie. `git tag -s -a api/v1.x.y -m api/v1.x.y`.
+  In this repository, those directories are `api` and `tests`.
+  This is enables the tag to be used as a Go module version for any downstream
+  users.
 
-#### Manual Process
+### Release notes
 
-1. Run `make release-notes` to create the release notes . Copy the output and sort
-   manually the items that need to be sorted.
-1. Create annotated tags:
+Next step is to clean up the release note manually.
 
-   ```bash
-   git tag -a -s v0.x.y -m v0.x.y
-   git tag -a -s test/v0.x.y -m test/v0.x.y
-   ```
+- Check for duplicates, reverts, and incorrect classifications of PRs, and
+  whatever release creation tagged to be manually checked.
+- For any superseded PRs (like CAPI uplifts, or commit revertions) that provide
+  no value to the release, create a summary line in `Superseded` section with
+  the PR ids and summary title. This way thye are acknowledged to be part of
+  the release, but not overwhelming the actual changes contained by the release.
+- Save the draft.
 
-   To use your GPG signature when pushing the tag, use `git tag -s [...]` instead
-1. Push the tags to the GitHub repository:
+### Tagging the Ironic and Mariadb images
 
-   ```bash
-   git push origin v0.x.y
-   git push origin test/v0.x.y
-   ```
+NOTE: This part will be moved to BMO releasing after BMO release process and
+maintenance plan has been agreed on.
 
-   NB: `origin` should be the name of the remote pointing to
-   `github.com/metal3-io/cluster-api-provider-metal3`
-1. Run `make release` to build artifacts (the image is automatically built by CI)
-1. [Create a release in GitHub](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository)
-   that contains the elements listed above that have been created in the `out`
-   folder
-1. Create a branch `release-0.x` for a minor release for backports and bug fixes.
-   This is not needed for patch releases.
+Before publishing CAPM3 release, we need to tag corresponding
```

Comment:
Not necessarily before capm3 release I would say, but once CAPM3 is published (at least in the past we used to do so)

Reply:
Why would you want to do them later? When you go public with a release, why'd you intentionally want to have it half-ready? If you face issues tagging the images, or Quay wouldn't build them etc, then your release is broken.

Reply:
I would not recommend tagging it prior CAPM3 tag pushing, since the tag that is used as a prefix for these are the same as CAPM3 tag you will be pushing and you may end up in a situation where users might get confused with these (ironic tags available but not CAPM3, due to some technical reason when you could not push tag in CAPM3). So, IMO it is better to carry out these steps post CAPM3 tag pushing so we are backed up and referencing something that already exists upstream

Reply:
I'd still argue that before you press the publish button, nobody needs to care (won't even notice really) is more correct approach than publish first, and then scramble. Publishing a release sends out emails to everyone watching repo, might even trigger an automated workflow somewhere, so it is much bigger failure do it that way.